pkg/manager/controllers/addonconfiguration/addon_configuration_reconciler.go

```diff
@@ -131,8 +137,9 @@ func (d *managedClusterAddonConfigurationReconciler) mergeAddonConfig(
 			}
 
 			match = true
-			if mergedConfigs[i].ConfigReferent != config.ConfigReferent {
+			if !equality.Semantic.DeepEqual(mergedConfigs[i].DesiredConfig, config.DesiredConfig) {
```

Comment:
what happens if configReferent is the same but desiredConfig is different? should we not clean the generation?

Reply:
yes, this part need modification, should keep the generation when configReferent (name/namespace) is the same and only config spec hash changes.

Reply:
fixed in latest commit.